src/PAModel/MergeTool/Deltas/ChangeProperty.cs

```diff
@@ -1,36 +1,78 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Microsoft.PowerPlatform.Formulas.Tools.EditorState;
 using Microsoft.PowerPlatform.Formulas.Tools.IR;
+using Microsoft.PowerPlatform.Formulas.Tools.Schemas;
 using Microsoft.PowerPlatform.Formulas.Tools.Utility;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Text;
 
 namespace Microsoft.PowerPlatform.Formulas.Tools.MergeTool.Deltas
 {
     internal class ChangeProperty : IDelta
     {
-        public ControlPath ControlPath;
-        public string PropertyName;
-        public string Expression;
-        public bool WasRemoved = false;
+        public readonly ControlPath ControlPath;
+        public readonly string PropertyName;
+        private readonly string _expression;
+        private readonly bool _wasRemoved;
+        private readonly CustomPropertyJson _customProperty;
+
+        // For a normal property that changed
```

Comment:
> // For a normal property that changed [](http://example.com/codeflow?start=7&length=38)

nit: comments are great, consider static builder methods as a means to distinguish different ctors. 

public static ChangeProperty NewPropertyChanged(...)
public static ChangeProperty NewCustomPropertyChanged(...)
public static ChangeProperty NewRemoved(...)